APE10.rst

```diff
@@ -0,0 +1,178 @@
+Roadmap for Python 3-only support
+---------------------------------
+
+author: Tom Robitaille
+
+date-created: 2016 July 13
+
+date-last-revised: 2016 July 13
+
+type: Process
+
+status: Draft
+
+Abstract
+--------
+
+The purpose of this APE is to set out a roadmap for future releases of the
+Astropy core package, in particular to identify the process for switching to
+releases that support only Python 3.
+
+Detailed description
+--------------------
+
+A number of scientific Python projects, including IPython, Matplotlib, Sympy,
+and others have now signed the `Python 3 statement
+<https://python3statement.github.io>`_ which states that scientific Python
+packages will drop support for Python 2 no longer than 2020. In fact, the
+IPython developers have now released `IPython 5.0
+<http://blog.jupyter.org/2016/07/08/ipython-5-0-released/>`_, which will be the
+last major release to support Python 2 - since it is a long term support (LTS)
+release, there will be 5.x bug fix releases until some time in 2019, but future
+major releases that add new features, starting with 6.0 (in 2017), will support
+only Python 3. This APE proposes that the Astropy project should also sign the Python 3 statement
+statement, and provides a target roadmap for this to happen.
+
+Roadmap
+-------
+
+The proposed roadmap for releases, including past releases for reference, is the
+following:
+
+==========   ==================  ==========================================
+Version      Release date        Notes
+==========   ==================  ==========================================
+v1.0 - LTS   February 19, 2015   Supported with bug fixes until June 2017
+v1.1         December 11, 2015
+v1.2         June 23, 2016
+v1.3         December 2016
+v2.0 - LTS   June 2017           Supported with bug fixes until end of 2019
+v3.0         December 2017       First release to support only Python 3+
+v3.1         June 2018
+v3.2         December 2018
+v3.3         June 2019
+v4.0 - LTS   December 2019       Supported with bug fixes for two years)
+==========   ==================  ==========================================
+
+Note that v1.0, v2.0, and v4.0 are marked as long-term support releases (LTS),
+which means that they are typically supported with bug fixes for at least two
+years. The above roadmap proposes that the last Python 2-compatible release will
+be v2.0, which we will support with bug fixes until the end of 2019. In the mean
+time, v3.0, to be released in December 2017, will be the first release that does
+not support Python 2.
+
+Note that v3.0 would not be an LTS release, since we would
+otherwise need to maintain two LTS releases in parallel. This is a one-off
+deviation from the version numbering policy outlined in `APE 2
+<https://github.com/astropy/astropy-APEs/blob/master/APE2.rst>`_.
```

Comment:
And maybe add something like: "and subsequent releases will follow the pattern established by APE 2.